### PR TITLE
Feature/final mutations

### PR DESCRIPTION
**Pull request by kevinhartmann23**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [x] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
*Briefly, but clearly describe what you did, and explain any new code.*
ADDS ALL MUTATIONS FOR MVP

## *Instructions for reviewer:*
*Where should the reviewer start?*
*How should this be manually tested?*
*Any background context you want to provide?*
*What are the relevant tickets?*

## *Issues:*
*Are there any concerns, issues, or bugs in this branch? If so describe them.*
subEvent Create does not return correct date format
onCompleted: is not rerendering state, I'm having a hard time approaching this.... Will tackle in the morning.

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: src/components/calendar/event-utils.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ export function formatEvents(eventsArray) {
         venue: event.venue,
         start: event.startTime,
         end: event.endTime,
+        subEvents: event.subEvents
```
Comment (kristenmb) [inline src/components/calendar/event-utils.js:16]: Its awesome that the calendar lets us add extended props!

File: src/components/agenda/Agenda.js [hunk 2]
```diff
@@ -3,15 +3,11 @@ import AgendaList from './AgendaList'
 import AgendaSideBar from './AgendaSideBar'
 import './Agenda.css'
 
-function Agenda({setAgenda, agenda}) {
-
-  const updateAgenda = (listItem) => {
-    setAgenda([...agenda, listItem])
-  }
+function Agenda({ agenda, currentEvent}) {
```
Comment (richardltyler) [inline src/components/agenda/Agenda.js:6]: I love how clean these components get as we introduce the global state

File: src/components/contacts/Contacts.js [hunk 3]
```diff
@@ -49,19 +50,26 @@ const Contacts = () => {
...
+    setFilterContacts(filteredContacts);
   }
 
   const toggleAddContactModule = (event) => {
     event.preventDefault();
     toggleIsAddingContact(!isAddingContact);
   }
 
-  const contactCards = contacts.map(contact => <ContactCard key={contact.id} contact={contact}/>);
+  useEffect(() => {
```
Comment (kristenmb) [inline src/components/contacts/Contacts.js:61]: This is a great addition

File: src/components/agenda/AgendaSideBar.js [hunk 4]
```diff
@@ -1,48 +1,34 @@
...
-    } else if (hours === 0) {
-      timeValue = "12";
-    }
-
-    timeValue += (minutes < 10) ? ":0" + minutes : ":" + minutes
-    timeValue += (hours >= 12) ? "PM" : "AM"
-
-    return timeValue
+    console.log("input time",time)
+    const date = currentEvent.extendedProps.start.replace(/T.*$/, '')
```
Comment (richardltyler) [inline src/components/agenda/AgendaSideBar.js:21]: regex ftw

**Review by kristenmb — APPROVED** [review]
This is such good work Kevin! Thank you for pushing through on this and busting all this API manipulation out.

File: src/components/contacts/ContactCard.js [hunk 5]
```diff
@@ -1,11 +1,18 @@
...
+import { useApp } from '../../context/AppContext'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faTrashAlt } from '@fortawesome/free-solid-svg-icons'
 
 const ContactCard = (props) => {
-  const { lastName, firstName, email, phone } = props.contact;
+  const { id, lastName, firstName, email, phoneNumber } = props.contact;
+  const { deleteContact } = useApp()
   return (
     <article className="ContactCard">
```
Comment (richardltyler) [inline src/components/contacts/ContactCard.js:10]: good eye for this delete functionality!

**Review by richardltyler — APPROVED** [review]
This looks awesome!

File: src/components/event/Event.js [hunk 6]
```diff
@@ -21,12 +27,14 @@ const Event = (props) => {
...
-  const saveAgenda = () => {
-    console.log(agenda)
-    //MUTATE SUBEVENTS HERE
+
+  const deleteSelectedEvent = (event) => {
+    event.preventDefault()
+    deleteEvent(currentEvent.publicId)
+    history.push({
+      pathname: '/calendar'
+    })
```
Comment (DrewBradley) [inline src/components/event/Event.js:36]: DESTROY

File: src/components/form/Form.js [hunk 7]
```diff
@@ -44,15 +47,49 @@ const Form = ({location}) => {
...
+      setSelectedVenue('NEW')
+    }
+    
+  }
+
+  const confirmEvent = (event) => {
+    event.preventDefault()
+   history.push({
+      pathname: '/calendar'
+    })
```
Comment (DrewBradley) [inline src/components/form/Form.js:79]: HISTORY!

**Review by DrewBradley — APPROVED** [review]

**Pull request merged**